script.js

```diff
@@ -1,4 +1,198 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
-};
+// Global variables
+let deck, hands, gameState, playerScore, dealerScore;
+
+// Helper function to create a deck of cards
+function createDeck() {
+  suits = ["♥", "♦", "♣", "♠"];
+  ranks = ["2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K", "A"];
+  deck = [];
+
+  var i = 0;
+  while (i < suits.length) {
+    var j = 0;
+    while (j < ranks.length) {
+      deck.push({ suit: suits[i], rank: ranks[j] });
+      j++;
+    }
+    i++;
+  }
+
+  return deck;
+}
+// Helper function to shuffle the deck of cards
+function shuffleDeck(deck) {
+  var i = deck.length - 1;
+  while (i > 0) {
+    var j = Math.floor(Math.random() * (i + 1));
+    [deck[i], deck[j]] = [deck[j], deck[i]];
+    i--;
+  }
+}
+
+// Helper function to deal two cards to each player
+function dealCards(deck) {
+  return {
+    player: [drawCard(deck), drawCard(deck)],
+    dealer: [drawCard(deck), drawCard(deck)],
+  };
+}
+
+// Helper function to calculate the score of a hand
+function calculateScore(hand) {
```

Comment:
Just noticed that you're using the function syntax, nice! I personally prefer this but again, syntax all depends the convention where your company is working at